src/hooks/useTransactions/useTransactions.ts

```diff
@@ -0,0 +1,147 @@
+import { ChainAdapter, Transaction } from '@shapeshiftoss/chain-adapters'
+import { useChainAdapters } from 'context/ChainAdaptersProvider/ChainAdaptersProvider'
+import { useWallet } from 'context/WalletProvider/WalletProvider'
+import dayjs from 'dayjs'
+import relativeTime from 'dayjs/plugin/relativeTime'
+import { fromBaseUnit } from 'lib/math'
+import { useCallback, useEffect, useState } from 'react'
+dayjs.extend(relativeTime)
+
+export type FormatTransactionType = Transaction & {
+  type: string
+  amount: string
+  date: string
+  dateFromNow: string
+}
+
+export type UseTransactionsReturnType = {
+  loading: boolean
+  txHistory: Record<string, FormatTransactionType[]>
+}
+
+export enum TxTypeEnum {
+  Sent = 'Sent',
+  Received = 'Received'
+}
+
+export enum TxStatusEnum {
+  Confirmed = 'confirmed',
+  Failed = 'failed'
+}
+
+export const getDate = (timestamp: number) =>
+  dayjs(Number(timestamp) * 1000).format('MM/DD/YYYY h:mm A')
+
+const formatTransactions = (txs: Transaction[], walletAddress: string): FormatTransactionType[] => {
+  return txs.map((tx: Transaction) => {
+    const date = getDate(tx.timestamp)
+    return {
+      ...tx,
+      type: walletAddress === tx.from ? TxTypeEnum.Sent : TxTypeEnum.Received,
+      amount: fromBaseUnit(tx.value, 18 /** TODO: get precision from asset service **/),
+      date,
+      dateFromNow: dayjs(date).fromNow(),
+      fee: fromBaseUnit(tx.fee, 18)
+    }
+  })
+}
+
+export const useTransactions = ({
+  chain = '',
+  contractAddress = '',
+  symbol = ''
+}: {
+  chain?: string | undefined
+  contractAddress?: string | undefined
+  symbol?: string | undefined
+} = {}): UseTransactionsReturnType => {
+  const [loading, setLoading] = useState<boolean>(false)
+  const [txHistory, setTxHistory] = useState<Record<string, FormatTransactionType[]>>({})
```

Comment:
Since these pieces of state are updated inside a resolved promise there is a possibility that a user could navigate away from where this is rendered and state could try to update when it is not mounted. I think that we should either...
1. Pull over our hooks from platform shared https://github.com/CroixDrinkers/platform-shared/blob/master/react/src/hooks/useStateIfMounted/useStateIfMounted.ts
2. look for a library that has some hooks for functionality like that
3. At least put TODOs in there to update to use `useStateIfMounted` hooks when we have it pulled over.

Reply:
Good call. I forgot about that. I'll pull over the `useStateIfMounted` hooks over because we'll need it

Reply:
Awesome you are the man!